src/components/Ragistration.jsx

```diff
@@ -0,0 +1,85 @@
+
+import React, {useState} from 'react';
+
+export const Registration = () => {
+  const [user, setUser] = useState({
+    email: '',
+    username: '',
+    password: '',
+    passwordConfirm: '',
+ })
+
+const handleInputChange = (id, value) => {
+ let name = "";
+  if(id === "email"){
+      setUser({...user, email: value});
+      if(value.includes("@") === true){
+      let indexOfAt = value.search("@") // vrací index na kterém je @
+      name = value.slice(0, indexOfAt)
+      console.log(value)
+      console.log(name)
+    } 
+      else { 
+      alert("Chybný formát emailu.")
+      }
+  }
+  if(id === "userName"){
+      setUser({...user, username: name}); // tady jsem to nevymyslela... :-(.. 
```

Comment:
co jsi tu nevymyslela? (resp, co jsi tady chtěla mít?)

Reply:
Snažila jsem se nastavit: Když vyplním email, tak aby se mi propsalo jméno do inputu User Name. To se mi sice povedlo, ale vždy mi zmizí vyplněný email. Pomocí spreadu se mi to nepodařilo, asi to použití nechápu.

Reply:
Už jsem na to přišla.